pkg/objectstate/defaulter/defaulter.go

```diff
@@ -0,0 +1,309 @@
+package defaulter
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	"github.com/openshift/apiserver-library-go/pkg/securitycontextconstraints/sccdefaults"
+)
+
+func Service(original, mutated client.Object) client.Object {
+	service := mutated.(*corev1.Service)
```

Comment:
do we need a check or do we want to panic if the object is not what we expect? applies to everything else below

Reply:
This code pulled directly from u/s we cannot change it if we want to keep it aligned with u/s

Reply:
> This code pulled directly from u/s we cannot change it if we want to keep it aligned with u/s

Sorry my bad it's not.

so basically we can, but the object is given by our code, meaning it's not provided from the API server or any other producer. 
We can log an error and return doing nothing. is that good enough?